tests/cpptests/test_cpp_query.cpp

```diff
@@ -491,15 +501,19 @@ TEST_F(QueryTest, testParser_v2) {
   assertInvalidQuery("*=>[KNN $K @vec_field $BLOB $EF ef foo bar x 5 AS score]", ctx); // parameter as attribute
   assertInvalidQuery("*=>[KNN $K @vec_field $BLOB EF ef foo bar x 5 AS ]", ctx); // not specifying score field name
   assertInvalidQuery("*=>[KNN $K @vec_field $BLOB EF ef foo bar x]", ctx); // missing parameter value (passing only key)
+  assertInvalidQuery("*=>[KNN $K @vec_field $BLOB => {$yield:dist}]", ctx); // invalid attributes syntax
+  assertInvalidQuery("*=>[KNN $K @vec_field $BLOB EF_RUNTIME 100 => {$yield_distance_as:dist;}]", ctx); // invalid combined syntax
```

Comment:
Why is it either? Why we can't mix?

Reply:
We can mix, the syntax error in this test is that the `]` of the KNN clause should have come before the `=>`